app/models/concerns/bulkrax/has_matchers.rb

```diff
@@ -100,42 +101,38 @@ def excluded?(field)
       mapping[field]['excluded'] || false
     end
 
-    def multiple_metadata?(name, node_name, node_content, object)
+    def multiple_metadata?(name, node_name, node_content, object_name)
       Rails.logger.info("Bulkrax Column automatically matched #{node_name}, #{node_content}")
       node_content = node_content.content if node_content.is_a?(Nokogiri::XML::NodeSet)
 
-      if object
-        parsed_metadata[object][name] ||= []
-        parsed_metadata[object][name] += node_content.is_a?(Array) ? node_content : Array.wrap(node_content.strip)
+      if object_name
+        parsed_metadata[object_name][name] ||= []
+        parsed_metadata[object_name][name] += node_content.is_a?(Array) ? node_content : Array.wrap(node_content.strip)
       else
         parsed_metadata[name] ||= []
         parsed_metadata[name] += node_content.is_a?(Array) ? node_content : Array.wrap(node_content.strip)
       end
     end
 
-    def matched_metadata?(matcher, multiple, name, node_content, object, object_multiple)
+    def matched_metadata?(matcher, multiple, name, index, node_content, object_name, object_multiple)
       result = matcher.result(self, node_content)
       return unless result
 
-      if object
+      if object_name
         if object_multiple
-          # find the index of the first object in the `object` array where the `name` key doesn't already exist
-          index = parsed_metadata[object].find_index { |obj| obj[name].nil? }
```

Comment:
setting the index this way no longer worked with the nested value because it would just create new objects instead of adding the value to existing key